### ISSUE TITLE
[Bug]: Autofill in other apps is not working

### ISSUE THREAD

**Issue by KazaHesto** [mozilla-mobile/fenix#24199]
### Steps to reproduce

1. Have a screen lock method set up (issue with no lockscreen setup is https://github.com/mozilla-mobile/fenix/issues/23707)
2. Have a login saved in Firefox
3. Set Firefox as the system autofill service (Settings -> Logins and passwords -> Autofill in other apps)
4. Open the app with a saved login in Firefox
5. Attempt to autofill the login with Firefox

### Expected behaviour

Login should autofill

### Actual behaviour

After unlocking (input pin/ fingerprint), nothing happens

### Device name

Samsung Galaxy S21 / Android Studio Emulator

### Android version

Android 12

### Firefox release type

Firefox

### Firefox version

98.1.1

### Additional information

Testing nightly builds from https://archive.mozilla.org/pub/fenix/nightly/ , the first bad build appears to be:

98.0a1 (Build #2015862031), bae20d650+
AC: 98.0.20220207143129, ec0201897a
GV: 98.0a1-20220206065831
AS: 91.0.1

2022-02-07T17:11:19.529657

Suspecting https://github.com/mozilla-mobile/android-components/commit/ec0201897ae2b2d91ec36be41986dff1c3192501 https://github.com/mozilla-mobile/android-components/issues/11666 since nothing else looks relevant in the range between the last good and first bad build

**Comment by Mugurell** [comment]
Tested with:
- OnePlus 7T with Android 11 and Pixel 3 with Android 12
- both using Gboard as keyboard - latest release from PlayStore
- current Firefox Release (98) and Nightly (100) as clean installs from PlayStore
- https://arstechnica.com/ , https://pinterest.com , https://twitter.com
- tried to use the logins in Firefox Focus (current release from PlayStore), the official Pinterest app (current release from Playstore) and the official Twitter app (current release from Playstore)
- used both pin and fingerprint as a device set security option.

Saw one scenario that worked:
- unlocking the credentials with both pin and password work
- login credentials do appear in the keyboard to be selected

https://user-images.githubusercontent.com/11428869/158336473-d702ae78-ee38-4600-985e-4d6a464a40e2.mp4

Continuing the steps from above:
- in some apps (the "Verification failed" dialog may be the indicator) a selected login doesn't autofill and then the autofill functionality doesn't work anymore (for that 3rd party app and autofill target)
- in other apps (maybe where the "Verification failed" dialog does not appear) the selected login does get autofilled.
(Saw off video in the pinterest app where the "Verification failed" dialog appears that here also the actual autofill doesn't work)


https://user-images.githubusercontent.com/11428869/158336970-4df36e24-fb9a-4272-976b-b1e5cf90ca31.mp4


---

It would seem like this second part is where the bug is: the initial authentication works and credentials are available but then the "Verification failed" agreement fails and credentials are not actually autofilled.

**Comment by Juriyx** [comment]
> After unlocking (input pin/ fingerprint), nothing happens

Hello. Passwords' unlock doesn't work on my Poco F3 with the same Android version as @Mugurell. A button with the fingerprint icon and "Unlock Firefox" text appears at the top of Gboard, I click on it and nothing happens. Touch to fingerprint on the right side of my smartphone also gives nothing. PIN code is set and fingerprints are added.
MIUI 12.5.6.0

Checked the latest versions (stable, beta and nightly) in Google Play. Is it a problem with the ROM on my device?

**Comment by DiegoARL38** [comment]
It doesn't work for me, either... I've got a Xiaomi Redmi Note 8 with MIUI 12.5.1.0 (Android 11) and I'm using TalkBack 12.1 as my screen reader.

The Unlock Firefox button appears, but nothing happens when tapped... I've tried having my finger in the fingerprint reader when tapping the unlock Firefox button, but no luck...

**Comment by Mugurell** [comment]
Issue to be fixed by https://github.com/mozilla-mobile/android-components/issues/11869.
We can retest related issue once that lands.
Thank you @KazaHesto for tracking the regression.

**Comment by Mugurell** [comment]
Fix should be available to test in the next Nightly.

**Comment by Mugurell** [comment]
> I checked, unlocking with pin does not work for me, neither does it work with the fingerprint.

This issue is just about the situation in which selecting a login doesn't actually autocompletes it - issue shown in the second video from https://github.com/mozilla-mobile/fenix/issues/24199#issuecomment-1067774300

Whether unlocking works or not seems like a separate issue which needs more investigations.
Can you please offer more details about the exact scenario you're testing? A video would help greatly (you don't have to use private websites or working logins).

**Comment by SoftVision-LorandJanos** [comment]
I checked on arstechnica.com and twitter.com with saved logins in the latest Nightly 100.0a1 (2022-03-17) with the issue still reproducible:

https://user-images.githubusercontent.com/48829350/158825148-3665a520-06e6-4d2a-b39e-43a6f89f96bb.mp4

**Comment by Mugurell** [comment]
@SoftVision-LorandJanos The scenario shown above is expected.
The same happens with Bitwarden: no autofill for username on twitter but working autofill for passwords:

https://user-images.githubusercontent.com/11428869/158826344-d0da669f-3e4c-4ecb-bc21-99d1b25b67cb.mp4

**Comment by DiegoARL38** [comment]
It's not fixed for me... After tapping Unlock Firefox, nothing happens...

* Firefox Nightly 100.0a1 2022-03-17T05:11:44.662421.
* Tested apps: Twitter, Reddit.
* Xiaomi Redmi Note 8 with MIUI 12.5.1.0 (Android 11 (.
* TalkBack version 12.1.

**Comment by Mugurell** [comment]
@DiegoARL38 This issue is just about the logins which if selected were not autofilled.
Issues with unlocking are still investigated in https://github.com/mozilla-mobile/fenix/issues/23707.
I would just have one question about unlocking: is there any screen lock set in device settings, or none?

**Comment by DiegoARL38** [comment]
@Mugurell Yes, I have got a pin set, and have enabled the option to unlock with a fingerprint, as well..

Okay, #23707 it is, then.

**Comment by KazaHesto** [comment]
The issue I had appears to be fixed, so closing the ticket
